### PR TITLE
ltx.is

### PR DESCRIPTION
**Pull request by sonnyp**
this is useful for es6 and you want to check something is an instance of Element but also for a more functional approach 

```javascript
import ltx from 'ltx'

ltx.is(foo)
```

instead of 

```javascript
import ltx from 'ltx'
const Element = {ltx}

foo instanceof Element || foo === 'string'
```

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -6,18 +6,6 @@
...
-      "name": "Astro",
-      "email": "astro@spaceboyz.net",
-      "web": "http://spaceboyz.net/~astro/"
-    },
-    {
-      "name": "Lloyd Watkin",
-      "email": "lloyd@evilprofessor.co.uk",
-      "web": "http://www.evilprofessor.co.uk"
-    }
-  ],
```
Comment (sonnyp) [inline package.json]: npm doesn't use those at all

File: lib/tag.js [hunk 2]
```diff
@@ -5,13 +5,12 @@ var parse = require('./parse')
 
 module.exports = function tag (/* [literals], ...substitutions */) {
   var literals = arguments[0]
-  var substitutions = Array.prototype.slice.call(arguments, 1)
 
   var str = ''
 
-  for (var i = 0; i < substitutions.length; i++) {
-    str += literals[i]
-    str += escape(substitutions[i])
```
Comment (sonnyp) [inline lib/tag.js]: unrelated but speed it up a little bit

**Comment by sonnyp** [comment]
@astro review?

File: lib/is.js [hunk 3]
```diff
@@ -0,0 +1,15 @@
+'use strict'
+
+var Element = require('./Element')
+
+module.exports.is = function is (el) {
+  return el instanceof Element || typeof el === 'string'
+}
```
Comment (sonnyp) [inline lib/is.js]: maybe isNode instead ?

**Comment by astro** [comment]
Although enlarging the API surface, convenience is good.

Once at it, I was thinking about other static methods: how about making the default export a shortcut to the Element constructor, like:
```js
module.exports = createElement
```

Oh wait, that's already used for tagged template literals. Does my proposition conflict with that?

**Comment by sonnyp** [comment]
`module.exports` is tag indeed but since it's not documented si we could change that.

Although I believe the current behavior to be the most convenient for most usage.

```javasript
const xml = ltx('<some xml="without user input"/>')

const xml = ltx`
  <some xml="${with user input}"/>
`
```

we already have a `ltx.createElement` method which is JSX compatible

```javascript
// use ltx.createElement pragma
const xml = <some xml="{with user input}"/>
```

File: lib/tag.js [hunk 4]
```diff
@@ -5,13 +5,12 @@ var parse = require('./parse')
...
+  for (var i = 1; i < arguments.length; i++) {
+    str += literals[i - 1]
+    str += escape(arguments[i])
```
Comment (sonnyp) [inline lib/tag.js]: unrelated but speed it up a little bit

**Comment by sonnyp** [comment]
I've renamed `ltx.is` --> `ltx.isNode`

**Comment by sonnyp** [comment]
# benchmark

## master

```
suite ltx
tag with template literal x 62,935 ops/sec ±1.43% (86 runs sampled)
tag with direct call x 65,062 ops/sec ±1.48% (89 runs sampled)
serialize and parse x 76,365 ops/sec ±1.54% (85 runs sampled)
createElement (jsx) x 554,031 ops/sec ±2.01% (87 runs sampled)
```

# this
```
suite ltx
tag with template literal x 69,788 ops/sec ±1.58% (86 runs sampled)
tag with direct call x 71,470 ops/sec ±1.79% (87 runs sampled)
serialize and parse x 72,761 ops/sec ±1.48% (87 runs sampled)
createElement (jsx) x 2,366,923 ops/sec ±5.82% (86 runs sampled)
```

File: lib/createElement.js [hunk 5]
```diff
@@ -13,10 +13,9 @@ var Element = require('./Element')
 module.exports = function createElement (name, attrs /*, child1, child2, ...*/) {
   var el = new Element(name, attrs)
 
-  var children = Array.prototype.slice.call(arguments, 2)
+  for (var i = 2; i < arguments.length; i++) {
+    el.cnode(arguments[i])
+  }
```
Comment (sonnyp) [inline lib/createElement.js]: x4 speedup

**Comment by sonnyp** [comment]
published as 2.4.0

**Pull request merged**